services/enterprise/utils/auth.py

```diff
@@ -80,10 +80,10 @@ def _decode_payload(self):
 
 class Authorize:
     def user(self, payload: dict) -> User:
-        email = payload[auth_settings.auth0_issuer + "email"]
-        user = user_service.get_user_by_email(email)
+        sub = payload['sub']
+        user = user_service.get_user_by_sub(sub)
         if not user:
-            raise UnauthorizedUserError(email=email)
+            raise UnauthorizedUserError(email=sub)
         return user
```

Comment:
In your PR comments you mentioned that it is backwards compatible but here you are completely removing the email authentication here.
There may be cases where the user signs up with simple username and password without using google auth or other third party authentication platform which I believe doesn't have the key work 'sub' in their header when they send a request.

Edit:
After some testing I believe using sub will probably be fine, I attempted to signup manually without google auth with 2 separate accounts and both had the `sub` key filled on the mongo database. I also looked through our existing prod database and found some users without `sub` which are probably added manually into the db. Therefore I will approve your PR